internal/clients/ec2/ec2_client.go

```diff
@@ -120,3 +120,34 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 
 	return resp.InstanceTypes, nil
 }
+
+func (c *Client) RunInstances(ctx context.Context, amount int32, instanceType types.InstanceType, imageId string, keyName string) ([]*string, *string, error) {
+	log.Trace().Msg("Run AWS EC2 instance")
+	input := &ec2.RunInstancesInput{
+		MaxCount:     aws.Int32(amount),
+		MinCount:     aws.Int32(amount),
+		InstanceType: instanceType,
+		ImageId:      aws.String(imageId),
+		KeyName:      &keyName,
+	}
+	resp, err := c.ec2.RunInstances(ctx, input, func(o *ec2.Options) {
+		o.Region = config.AWS.Region
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot run instances: %w", err)
+	}
+	instances := c.parseRunInstancesResponse(resp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot ParseRunInstancesResponse: %w", err)
+	}
+	return instances, resp.ReservationId, nil
+}
```

Comment:
Please add some logging as I described in other comment, I feel like it is very important because sooner or later a customer will come with "hey my instance id was spawned incorrectly".

Reply:
@lzap Should I add the logging here? 
I wanted to add it after creating the record: 
```golang
	// For each instance that was created in AWS, add it as a DB record
	for _, instanceId := range instances {
		err = resD.CreateInstance(ctx, &models.InstancesReservation{
			ReservationID: args.ReservationID,
			InstanceID:    *instanceId,
		})
		if err != nil {
			return fmt.Errorf("cannot create InstancesReservation for id %d: %w", instanceId, err)
		}
		logger.Info().Str("instance_id", *instanceId).Msgf("Created new instance via AWS reservation %s", *awsReservationId)
	}
```

Reply:
You got it right, I just commented here, at the wrong place.